edb/server/compiler/compiler.py

```diff
@@ -626,9 +626,15 @@ def _compile_ql_query(
         )
 
         if ir.cardinality.is_single():
-            result_cardinality = enums.Cardinality.AT_MOST_ONE
+            if ir.cardinality.can_be_zero():
```

Comment:
I would factor this into a classmethod on the enum similarly to how we convert between cardinality representations elsewhere:

```
    @classmethod
    def from_ir_value(
        cls,
        card: ir.Cardinality
    ) -> Cardinality:
        ...
```

then, use the new helper from other such open-coded conversion places (e.g. `cardinality_from_ptr`)